nixos/modules/services/misc/mesos-slave.nix

```diff
@@ -43,6 +51,12 @@ in {
         type = types.bool;
       };
 
+      withDocker = mkOption {
```

Comment:
@ragnard yes you can achieve both of those requirements using `config.virtualisation.docker.enable`. What's the status of this pull request? Still working on this? This is something I'm interested in as well.